### PR TITLE
feat(api)!: remove legacy hex hash field from storage responses

### LINKED ISSUE #110: Remove the legacy hex `hash` field from storage API responses
### Problem

#106 shipped `digestMultibase` as an additional field alongside the legacy hex SHA-256 `hash` field, deliberately keeping both during a transition period. Carrying two fields that name the same value pushes consumers to pick one and confuses the API surface: the legacy hex form has none of the self-describing properties (algorithm, encoding) that `digestMultibase` was introduced to provide.

The hex `hash` field also discourages the homebrew transcoding logic (`toDigestMultibase(hexHash)`) that PR #107 added to bridge the two — that logic is dead weight now that `@uncefact/untp-utils@0.1.0` (the canonical multibase-multihash primitive) is published and can compute the digest directly from input bytes.

### Current Behaviour

```json
{
    "uri": "http://localhost:3333/api/3.x.x/documents/2ad789c7-e513-4523-a826-ab59e1c423cd.json",
    "hash": "d6bb7b579925baa4fe1cec41152b6577003e6a9fde6850321e36ad4ac9b3f30a",
    "digestMultibase": "zQmcnsmRVVuPbmPwesYza9zXSbn5GJMQU4x9RnFDAZdcKCD"
}
```

`CryptographyService.computeHash` returns hex; a separate `toDigestMultibase` method transcodes hex bytes to base58btc multihash. The interface and the response carry both representations.

### Expected Behaviour

```json
{
    "uri": "http://localhost:3333/api/3.x.x/documents/2ad789c7-e513-4523-a826-ab59e1c423cd.json",
    "digestMultibase": "zQmcnsmRVVuPbmPwesYza9zXSbn5GJMQU4x9RnFDAZdcKCD"
}
```

- The `hash` field is removed from all storage response shapes (public + private, JSON + binary).
- `CryptographyService.computeHash` returns the multibase-encoded multihash directly, delegating to `@uncefact/untp-utils/multibase-digest`.
- `toDigestMultibase(hexHash)` and the direct `multiformats` dependency are dropped.
- Swagger, README, developer-guide, and how-it-works docs all stop describing the hex `hash` field.

### Impact

Breaking change for API consumers reading the `hash` field. UNTP reference-implementation consumers reading `digestMultibase` are unaffected (already supported during the transition). The UNTP storage adapter is being updated in its own repo to align with the canonical `digestMultibase` name.

### Workaround

Consumers still relying on the hex field can recompute it locally by base58btc-decoding the multibase digest and stripping the two-byte multihash prefix (`0x12 0x20` for sha2-256). Long-term, callers should adopt the multibase form.

Related to #106

### PR DESCRIPTION
**Pull request by ashleythedeveloper**
This PR drops the legacy hex SHA-256 `hash` field from all storage API responses, leaving `digestMultibase` as the sole content-integrity field. `CryptographyService` now delegates the multibase-multihash computation to `@uncefact/untp-utils@^0.1.0` instead of carrying its own transcoder, and the in-repo `multiformats` direct dependency is removed.

Builds on #107 (which added `digestMultibase` alongside `hash`) and #109 (which documented it); this PR completes the cutover.

**Breaking change**: API consumers reading `hash` must switch to `digestMultibase`. The UNTP storage adapter is being updated in its own repo to align with the canonical name; coordinate roll-out with that repo before deploying.

Closes #110

## Test plan
- [x] Unit tests pass (212 across 17 suites)
- [x] New `crypto.test.ts` pins known-vector digests for sha2-256 + sha2-512 and both base58btc + base64 encodings, plus the empty-input edge case
- [x] e2e helper recomputes `digestMultibase` via the same library and asserts byte-for-byte equality against the API response (JSON + binary paths)
- [x] Webpack production build succeeds
- [x] Lint clean on changed files
- [ ] Full docker e2e suite (`yarn test:e2e`) deferred to CI

### CODE REVIEW

File: src/services/cryptography/crypto.ts [hunk 1]
```diff
@@ -1,28 +1,75 @@
...
-    }
-
-    toDigestMultibase(hexHash: string): string {
-        const hashBytes = Buffer.from(hexHash, 'hex');
-        if (hashBytes.length !== 32) {
-            throw new Error(`Invalid SHA-256 hash length: expected 32 bytes, got ${hashBytes.length}`);
+    async computeHash(input: string | Buffer, options: IComputeHashOptions = {}): Promise<string> {
+        const algorithm = options.algorithm ?? HashAlgorithm.SHA256;
+        const base = options.base ?? MultibaseEncoding.BASE58_BTC;
+        const bytes = typeof input === 'string' ? new TextEncoder().encode(input) : new Uint8Array(input);
```
[the remark this answers is not in the document]
Comment (ashleythedeveloper) [reply src/services/cryptography/crypto.ts]: Done in 50c79d4 — passed Buffer through directly; `new Uint8Array(input)` was a defensive copy that didn't earn its keep for read-only hashing.

File: e2e/helpers.ts [hunk 2]
```diff
@@ -28,11 +28,29 @@ export function resolveUri(uri: string): string {
...
+ * by CryptographyService (sha2-256, base58btc).
+ *
+ * @see https://github.com/multiformats/multihash Multihash specification
+ * @see https://github.com/multiformats/multibase Multibase specification
  */
-export function computeHash(input: string | Buffer): string {
-    return crypto.createHash('sha256').update(input).digest('hex');
+export async function computeHash(input: string | Buffer): Promise<string> {
+    const { MultibaseDigest } = await importEsm<MultibaseDigestModule>('@uncefact/untp-utils/multibase-digest');
+    const bytes = typeof input === 'string' ? new TextEncoder().encode(input) : new Uint8Array(input);
```
[the remark this answers is not in the document]
Comment (ashleythedeveloper) [reply e2e/helpers.ts]: Done in 50c79d4 — same fix as the service-side comment, passing Buffer through directly.

File: src/services/cryptography/crypto.ts [hunk 3]
```diff
@@ -1,28 +1,75 @@
...
+        const { MultibaseDigest } = await loadMultibaseDigest();
+        try {
+            const digest = await MultibaseDigest.fromData(bytes, { algorithm, base });
+            return digest.toString();
+        } catch (err) {
+            console.error(
+                `[CryptographyService.computeHash] MultibaseDigest.fromData failed (algorithm=${algorithm}, base=${base}, bytes=${bytes.byteLength}).`,
+                err,
+            );
```
[the remark this answers is not in the document]
Comment (ashleythedeveloper) [reply src/services/cryptography/crypto.ts:74]: Not addressing — `console.error`/`console.info` is the established logging pattern in this codebase (see `src/routes/{public,private}/service.ts` and `src/services/storage/aws.ts`); introducing a dedicated logging abstraction is out of scope for this PR. Letting the error propagate without logging would also lose the algorithm/base/byte-length context, which is the diagnostic value the line carries.

File: src/services/cryptography/crypto.ts [hunk 4]
```diff
@@ -1,28 +1,78 @@
...
+        const algorithm = options.algorithm ?? HashAlgorithm.SHA256;
+        const base = options.base ?? MultibaseEncoding.BASE58_BTC;
+        const bytes = typeof input === 'string' ? new TextEncoder().encode(input) : input;
+        const { MultibaseDigest } = await loadMultibaseDigest();
+        try {
+            const digest = await MultibaseDigest.fromData(bytes, { algorithm, base });
+            return digest.toString();
+        } catch (err) {
+            console.error(
+                `[CryptographyService.computeDigestMultibase] MultibaseDigest.fromData failed (algorithm=${algorithm}, base=${base}, bytes=${bytes.byteLength}).`,
```
[the remark this answers is not in the document]
Comment (ashleythedeveloper) [reply src/services/cryptography/crypto.ts]: Done in 562c506 — moved `algorithm`, `base`, and `bytes` out of the template literal into a structured second argument to `console.error`. JS template literals are not printf-style so the values were never reinterpreted as format directives in practice, but the structured-args form satisfies the check and reads better as structured logging.
